sympy/concrete/summations.py

```diff
@@ -419,6 +419,91 @@ def reverse_order(self, *indices):
 
         return Sum(e * self.function, *limits)
 
+    def findrecur(self, F=Dummy('F'), n=None):
```

Comment:
I think `Dummy('F')(x)` just creates `Function('F')(x)`, and there is no DummyFunction (https://github.com/sympy/sympy/issues/6465). We might as well just make this `F=Function('F')` to remove confusion.